Source/WebCore/style/StyleBuilder.cpp

```diff
@@ -401,7 +401,7 @@ void Builder::applyProperty(CSSPropertyID id, CSSValue& value, SelectorChecker::
     BuilderGenerated::applyProperty(id, m_state, valueToApply.get(), valueType);
 
     if (!isUnset) {
-        if (cascadeLevel == CascadeLevel::Author && m_state.element()->isDevolvableWidget() && CSSProperty::disablesNativeAppearance(id) && m_state.applyPropertyToRegularStyle())
+        if (cascadeLevel != CascadeLevel::UserAgent && CSSProperty::disablesNativeAppearance(id) && m_state.applyPropertyToRegularStyle())
```

Comment:
Note that the spec explicitly mentions the author origin: https://drafts.csswg.org/css-ui/#appearance-disabling-properties

Though this does seem like a case where the spec should probably be updated, I don't think it makes sense to exclude the User Origin from this behavior, as we do right now.

Reply:
I filed https://github.com/w3c/csswg-drafts/issues/11555.